data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java

```diff
@@ -645,4 +674,56 @@ public int count() {
     public boolean isNull(int columnIndex) {
         return get(columnIndex) == null;
     }
+
+    /**
+     * Helper method to normalize malformed datetime strings to make them parseable by Instant.parse()
+     * This handles common malformations like single-digit seconds, missing digits, etc.
+     */
+    private String normalizeMalformedDateTime(String dateString) {
```

Comment:
this is added to handle 2023-07-15T14:30:4Z edge cases(KustoResultSetTableDateTest->testGetDate_WithEdgeCaseLengths_HandlesCorrectly()) which is parsed properly in the FastDateFormat.

Reply:
Can you expand more? How can we get this number? Or was this test just always there?

Reply:
changed the implementation a bit to use flexible format using a formatter builder. Can you have a look on this @AsafMah

Reply:
Fixed with DateTimeformatter builder and tests seem to pass